python/nav/mibs/mibretriever.py

```diff
@@ -251,19 +251,19 @@ def __init__(cls, name, bases, dct):
         # modify mib data to slightly optimize later OID manipulation
         convert_oids(mib)
 
-        MibRetrieverMaker.__make_node_objects(cls)
+        MibRetrieverMaker.__make_node_objects()
         cls.tables = dict((t.table.name, t)
                           for t in MibTableDescriptor.build_all(cls))
 
-        MibRetrieverMaker.__make_scalar_getters(cls)
-        MibRetrieverMaker.__make_table_getters(cls)
+        MibRetrieverMaker.__make_scalar_getters()
+        MibRetrieverMaker.__make_table_getters()
 
         MibRetrieverMaker.modules[mib['moduleName']] = cls
 
     # following is a collection of helper methods to modify the
     # MIB-aware retriever class that is being created.
 
-    @staticmethod
+    @classmethod
     def __make_scalar_getters(cls):
         """Make a get_* method for every scalar MIB node."""
```

Comment:
Yeah, this is a little bizarre. The point of a staticmethod decorator is that you don't pass in the class, and the point of a staticmethod in python is that you do not *depend* on anything in the class.

Reply:
The confusion may stem from using `cls` as the variable name here, since this is not a reference to the MibRetrieverMaker class, but the class that is being built by the MibRetrieverMaker metaclass.

Reply:
Perhaps `klass` would be a better moniker in these cases.